### PR TITLE
Issue 21380

### PR DESCRIPTION
**Pull request by bmukheja**
Added a visualisation, and made changes to the dashboard as necessary. Now the apps would be displayed in a pie chart, and limited by an upper number of 10.

### CODE REVIEW

File: public/configurations/visualizations/aar-enterprise-app-pie.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        "pieOuterRadius": 0.95,
+        "legend": {
+            "show": true,
+            "orientation": "horizontal",
+            "circleSize": 1,
+            "labelOffset": 2
+        }
+    },
+    "listeners": [{
+      "redirect": "/dashboards/dateHistogramExample",
```
Comment (ronakmshah) [inline public/configurations/visualizations/aar-enterprise-app-pie.json]: What is this for?

File: public/configurations/visualizations/aar-enterprise-app-pie.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      "params": {
+          "app": "L7Classification"
+      }
+    }],
+    "query": "aar-enterprise-app-vertical-bar"
```
Comment (ronakmshah) [inline public/configurations/visualizations/aar-enterprise-app-pie.json]: Use the same name as in visualization

File: public/configurations/visualizations/aar-enterprise-app-pie.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+    "id": "aar-enterprise-app-pie",
```
Comment (ronakmshah) [inline public/configurations/visualizations/aar-enterprise-app-pie.json]: Name of this file should be "aar-enterprise-all-app-usage.json" or something which is more meaningful

File: public/configurations/visualizations/aar-enterprise-app-pie.json (hunk 2 above)
Comment (ronakmshah) [inline public/configurations/visualizations/aar-enterprise-app-pie.json]: Where is the change in the dashboard "aarEnterprise" where the current visualization is replaced by this one?

File: public/configurations/visualizations/aar-enterprise-app-pie.json (hunk 1 above)
Comment (bmukheja) [reply public/configurations/visualizations/aar-enterprise-app-pie.json]: Corrected it now. The query which I was replicating here had this originally pointed, changed to enterprisedetail as in the previous visualisation.

**Comment by ronakmshah** [comment]
@bmukheja the base branch has been merged to master. Percentage functionality talked about is being implemented and its a default type when displaying "other". So you need to change your "other" field to cover up to 85% with individual apps and devote 15% to others. Please rebase this master, make the change and refresh this pull.

**Comment by bmukheja** [comment]
Updated the pull request with 85% limit

File: public/configurations/dashboards/test_bharat.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (ronakmshah) [inline public/configurations/dashboards/test_bharat.json]: Please remove this file

File: public/configurations/queries/aar-enterprise-nsg-app-cumulative.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                        {
+                            "range": {
+                                "timestamp": {
+                                    "gte": "{{startTime:now-24h}}",
+                                    "lte": "{{endTime:now}}",
+                                    "format": "epoch_millis"
+                                }
+                            }
+                        },
+                        {
```
Comment (ronakmshah) [inline public/configurations/queries/aar-enterprise-nsg-app-cumulative.json]: Why below condition is required?

File: public/configurations/queries/aar-enterprise-all-app-usage.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                                    "format":"epoch_millis"
+                                }
+                            }
+                        }
+                    ]
+                }
+            },
+            "aggs":{
+                "3":{
+                    "filters":{
```
Comment (ronakmshah) [inline public/configurations/queries/aar-enterprise-all-app-usage.json]: Remove this aggregation and instead add the term check to the bool query above.

File: public/configurations/queries/aar-enterprise-nsg-app-cumulative.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                            "term": {
+                                "L7ClassEnhanced":"{{appName}}"
+                            }
+                        }
+                    ]
+                }
+            },
+             "aggs": {
+                "SourceNSG": {
```
Comment (ronakmshah) [inline public/configurations/queries/aar-enterprise-nsg-app-cumulative.json]: This should actually be a term query coming from nsg-per-app graph. And you need to aggregate users instead

**Comment by bmukheja** [comment]
Recreated branch bmukheja/21380 to apply git rebase

**Pull request closed without merging**